### PR TITLE
feature/fill-filterOutput

### PR DESCRIPTION
**Pull request by saminahbab**
### What
The work in this PR is in relation to the ticket [here](https://trello.com/c/r5UZqMLJ/5668-update-submitfilter-to-create-full-filteroutput-entry-in-mongo)
- Broke off filterOutput struct generation to own function
- Padded out the struct with the appropriate information 
- Set the Kafka producer to log errors 
   - This was because the `filters/ID/submit` endpoint has been observed timing out in develop but working locally. This could help with debugging the endpoint in test environments.
- Amended component tests to ensure that the appropriate filterOuput is persisted to the document store. 

### How to review

run `make test` and `make test-component`

### Who can review

Anyone from team B

### CODE REVIEW

**Comment by PauloPortugal** [comment]
@saminahbab please rebase `develop` into your branch to sort out the code conflict

**Comment by PauloPortugal** [comment]
What is the ticket this PR relates to?
This helps the reviewer understand if the goals/requirements were met.

**Comment by saminahbab** [comment]
> What is the ticket this PR relates to? This helps the reviewer understand if the goals/requirements were met.
@PauloPortugal description now contains the ticket, cheers

**Comment by saminahbab** [comment]
> @saminahbab please rebase `develop` into your branch to sort out the code conflict

@PauloPortugal rebased `develop` into my branch, conflict managed.

File: api/filters.go [hunk 1]
```diff
@@ -139,14 +139,9 @@ func (api *API) submitFilter(w http.ResponseWriter, r *http.Request) {
 
 	}
 
-	filterOutput := &model.FilterOutput{
-		FilterID:   filter.ID,
-		State:      model.Submitted,
-		Events:     []model.Event{},
-		Dimensions: []model.Dimension{},
-	}
+	filterOutput := api.generateFilterOutput(filter)
```
Comment (ChrysmOre) [inline api/filters.go:142]: I don't know if I like the idea of creating a function for this, the whole thing could be done in one statement.

Comment (saminahbab) [reply api/filters.go:142]: I'm happy to change that

**Review by ChrysmOre — APPROVED** [review]
LGTM

**Pull request merged**